packages/headless/src/features/commerce/facets/facet-set/facet-set-slice.test.ts

```diff
@@ -932,20 +991,294 @@ describe('commerceFacetSetReducer', () => {
     });
   });
 
-  // TODO describe('for hierarchical facets', () => { /* ... */ });
+  describe('for category facets', () => {
```

Comment:
Here I'm essentially adapting the toggleSelectCategoryFacetValue tests from category-facet-set-slice.ts; I rephrased some descriptions to make things clearer in places, and did a bit of restructuring, but overall the logic is the same.